Tests/Packet++Test/Tests/WireGuardTests.cpp

```diff
@@ -0,0 +1,336 @@
+#include "../TestDefinition.h"
+#include "../Utils/TestUtils.h"
+#include "Packet.h"
+#include "WireGuardLayer.h"
+#include "SystemUtils.h"
+#include <cstring>
+#include "EndianPortable.h"
+
+PTF_TEST_CASE(WireGuardHandshakeInitParsingTest)
+{
+	timeval time;
+	gettimeofday(&time, nullptr);
+
+	READ_FILE_AND_CREATE_PACKET(1, "PacketExamples/WireGuardHandshakeInitiation.dat");
+
+	pcpp::Packet wgHandShakeInitPacket(&rawPacket1);
+
+	PTF_ASSERT_TRUE(wgHandShakeInitPacket.isPacketOfType(pcpp::WireGuard));
+
+	pcpp::WireGuardLayer* wgLayer = wgHandShakeInitPacket.getLayerOfType<pcpp::WireGuardLayer>();
+	PTF_ASSERT_NOT_NULL(wgLayer);
+
+	pcpp::WireGuardHandshakeInitiationLayer* wgHandShakeInitLayer =
+	    wgHandShakeInitPacket.getLayerOfType<pcpp::WireGuardHandshakeInitiationLayer>();
+	PTF_ASSERT_NOT_NULL(wgHandShakeInitLayer);
+
+	PTF_ASSERT_EQUAL(wgHandShakeInitLayer->getMessageTypeAsString(), "Handshake Initiation");
+	PTF_ASSERT_EQUAL(wgHandShakeInitLayer->toString(),
+	                 "WireGuard Layer, " + wgHandShakeInitLayer->getMessageTypeAsString() + " message");
+
+	PTF_ASSERT_TRUE(wgHandShakeInitLayer->getWireGuardMessageType() ==
+	                pcpp::WireGuardLayer::WireGuardMessageType::HandshakeInitiation);
+
+	PTF_ASSERT_EQUAL(wgHandShakeInitLayer->getSenderIndex(), be32toh(818952152));
```

Comment:
no need to use `be32toh` here, we can just put the value from Wireshark: `0x30d037d8`